### PR TITLE
feat(agent): Fix GKE Autopilot installation

### PR DESCRIPTION
**Pull request by aroberts87**
## What this PR does / why we need it:
There are several items related to the Agent chart that do not work when installing into a GKE Autopilot cluster. This change does the following to address them:

- Set a 500Mi ephemeral storage request on the agent container so that the containers are guaranteed enough space to compile the BPF probe
- Either take a priority class name, or optionally create a new priority class, to ensure the Agent pods get scheduled on the initial set of small nodes provided by Autopilot
- Enforce the use of the regular (non-slim) agent container in GKE Autopilot 

## Checklist

- [x] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [x] Chart Version bumped for the respective charts
- [x] Variables are documented in the README.md (or README.tpl in some charts)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

### CODE REVIEW

**Review by saltycr3w — APPROVED** [review]
Looks good, and it's configurable, yay!

Did you you have to move the merge into the daemonset.yaml instead of inside the _helper for a particular reason?  Just curious.

**Review by dtaracks — APPROVED** [review]
So much better now - Thanks

**Comment by aroberts87** [comment]
> Looks good, and it's configurable, yay!
> 
> Did you you have to move the merge into the daemonset.yaml instead of inside the _helper for a particular reason? Just curious.

@dthomson-sysdig - Are you talking about for the kmodule container? I left it in the daemonset instead of moving to the _helper file because those values were just being taken directly from the `values.yaml` and don't ever need to comply with the resource profiles, so I figured it would reduce the complexity to just handle the splicing locally. Does that make sense?

File: charts/agent/templates/daemonset.yaml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {{- if .Values.daemonset.deploy }}
+{{- if and .Values.gke.autopilot .Values.slim.enabled }}
```
Comment (mavimo) [inline charts/agent/templates/daemonset.yaml]: use `{{- include "agent.gke.autopilot" . }}` to check for autopilot using the helper.

File: charts/agent/templates/daemonset.yaml [hunk 2]
```diff
@@ -29,8 +32,8 @@ spec:
       {{- end }}
     spec:
       serviceAccountName: {{ template "agent.serviceAccountName" .}}
-{{- if .Values.priorityClassName }}
-      priorityClassName: "{{ .Values.priorityClassName }}"
+{{- if or .Values.priorityClassName .Values.gke.autopilot }}
```
Comment (mavimo) [inline charts/agent/templates/daemonset.yaml]: use `{{- include "agent.gke.autopilot" . }}` to check for autopilot using the helper.

File: charts/agent/templates/gkeautopilotpriorityclass.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{{- if and .Values.gke.autopilot (not .Values.slim.enabled) }}
```
Comment (mavimo) [inline charts/agent/templates/gkeautopilotpriorityclass.yaml]: use `{{- include "agent.gke.autopilot" . }}` to check for autopilot using the helper.

File: post-change-3.yaml [hunk 4]
```diff
@@ -0,0 +1,469 @@
+---
```
Comment (aroberts87) [inline post-change-3.yaml]: Was this file intended to be committed?

File: pre-change-3.yaml [hunk 5]
```diff
@@ -0,0 +1,486 @@
+---
```
Comment (aroberts87) [inline pre-change-3.yaml]: Same as above, was this file intended to be committed?

File: charts/agent/templates/gkeautopilotpriorityclass.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{{- if and (include "agent.gke.autopilot" .) (not .Values.slim.enabled) .Values.gke.createPriorityClass }}
```
Comment (aroberts87) [inline charts/agent/templates/gkeautopilotpriorityclass.yaml]: Should we remove the constraint around slim not being enabled now?

File: charts/agent/templates/deployment.yaml [hunk 7]
```diff
@@ -105,28 +102,13 @@ spec:
...
-              readOnly: true
-            {{- else }}
-            - mountPath: /host/etc
-              name: etc-vol
-              readOnly: true
-            {{- end }}
+          {{- /* Always requested */}}
             - mountPath: /host/dev
               name: dev-vol
+              readOnly: false
```
Comment (aroberts87) [inline charts/agent/templates/deployment.yaml]: What is the reason for this change?

File: charts/agent/templates/daemonset.yaml [hunk 8]
```diff
@@ -72,32 +72,29 @@ spec:
...
       {{- end }}
       {{- if .Values.slim.enabled }}
       initContainers:
         - name: sysdig-agent-kmodule
           image: {{ template "agent.image.kmodule" . }}
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           securityContext:
-            capabilities:
-              drop:
-                - ALL
```
Comment (aroberts87) [inline charts/agent/templates/daemonset.yaml:85]: What is the purpose of this change?

File: charts/agent/templates/daemonset.yaml [hunk 9]
```diff
@@ -112,45 +109,51 @@ spec:
               value: {{ .Values.proxy.noProxy | default .Values.global.proxy.noProxy }}
           {{- end }}
           volumeMounts:
+          {{- /* Slim = true, Autopilot = false */}}
+          {{- if not (include "agent.gke.autopilot" .) }}
```
Comment (aroberts87) [inline charts/agent/templates/daemonset.yaml:113]: We're not checking the value of `slim.enabled` here, so won't this be picked in both cases?

File: charts/agent/templates/daemonset.yaml [hunk 10]
```diff
@@ -112,45 +109,51 @@ spec:
...
         - name: sysdig
           image: {{ template "agent.image" . }}
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           resources:
-{{ include "agent.resources" . | indent 12 }}
+            {{- include "agent.resources" . | nindent 12 }}
           securityContext:
-            capabilities:
-              drop:
-                - ALL
```
Comment (aroberts87) [inline charts/agent/templates/daemonset.yaml:153]: What is the reason for this change?

File: charts/agent/templates/daemonset.yaml (hunk 8 above)
Comment (mavimo) [reply charts/agent/templates/daemonset.yaml:85]: We're specifying this need to run as a privileged container, so the capabilities drop won't have any effect.

File: charts/agent/templates/deployment.yaml (hunk 7 above)
Comment (mavimo) [reply charts/agent/templates/deployment.yaml]: Is the default value, I added it supposing that make it more clear, but let's remove it 😅

File: charts/agent/templates/daemonset.yaml (hunk 9 above)
Comment (mavimo) [reply charts/agent/templates/daemonset.yaml:113]: Is by default since the init container is available only in slim mode (see [line 77](https://github.com/sysdiglabs/charts/pull/975/files#diff-18888d7372d81c06ea1ec38c2606ae16bf04d0248bdb8ad475cbc5cc22bbc364R77))

File: charts/agent/templates/daemonset.yaml (hunk 10 above)
Comment (mavimo) [reply charts/agent/templates/daemonset.yaml:153]: We're specifying this need to run as a privileged container, so the capabilities drop won't have any effect.
